### DISCUSSION TITLE
Release v0.22.0 Beta

### DISCUSSION THREAD

**Discussion by PrestonN** [FreeTubeApp/FreeTube#5948] [Announcements]
## Summary

Our Video Player has undergone a massive overhaul. We have switched from video.js to shaka-player. This allows us to add allot of new features and fix many issues. Notably, it's now possible to watch higher quality video streams. A massive thank you to @absidue for their help and support on this!

This is a very large change to how we handle things. If you happen to see anything that seems broken compared to the last release, please consider commenting about it in the thread below so that we can look at it.

For users hoping to import their subscriptions from YouTube, the process should be much quicker and a lot smoother since we no longer fetch thumbnail information when importing from YouTube. The channel thumbnails in the sidebar will be populated as you navigate around the app.

We would also like to thank all the translators that make FreeTube more accessible to all of our users!

> [!NOTE]
> The setting `Export Playlists For Older FreeTube Versions` will be deprecated in v0.24.0. This setting was introduced in v0.20.0 when we launched our full-fledged User Playlists. This allowed users to export their playlists into one big playlist and import the file into FreeTube versions older than v0.20.0. This setting is being deprecated because we don't recommend sticking with older versions of FreeTube. It is highly recommended that you upgrade to the latest version of FreeTube ff you are still using an outdated version of FreeTube.

> [!NOTE]
> This will be the last release to support macOS 10.15, as Electron v33.0.0 does not support this OS version. MacOS 11 (Big Sur) or later will be required to run future versions of FreeTube.

## Features
* Migrate video player from video.js to shaka-player by @absidue in https://github.com/FreeTubeApp/FreeTube/pull/4978
* Local API: Use IOS HLS manifest for livestreams by @absidue in https://github.com/FreeTubeApp/FreeTube/pull/5705
* Add a videos tab to artist topic channels by using YouTube's auto-generated playlists on the local API by @absidue in https://github.com/FreeTubeApp/FreeTube/pull/5661
* Implement persistent subscription cache by @PikachuEXE in https://github.com/FreeTubeApp/FreeTube/pull/5185
* Add ability to see comment replies of community posts through Invidious by @ChunkyProgrammer in https://github.com/FreeTubeApp/FreeTube/pull/5589
* Display video tags in search results with the Invidious API by @ChunkyProgrammer in https://github.com/FreeTubeApp/FreeTube/pull/5590
* Add support for Invidious instances that require the Basic authentication scheme to access them by @absidue in https://github.com/FreeTubeApp/FreeTube/pull/5569
* Implement case sensitive search in history by @PikachuEXE in https://github.com/FreeTubeApp/FreeTube/pull/5446
* Proxy live streams when "Proxy Videos Through Invidious" is enabled by @absidue in https://github.com/FreeTubeApp/FreeTube/pull/5649
* Add support for copying hashtag and post links from the context menu by @absidue in https://github.com/FreeTubeApp/FreeTube/pull/5538 and https://github.com/FreeTubeApp/FreeTube/pull/5760
* Add a GPU Internals entry to the View menu by @absidue in https://github.com/FreeTubeApp/FreeTube/pull/5550
* Add support for Belarusian locale by @aliaksandr-trush in https://github.com/FreeTubeApp/FreeTube/pull/5841
* Add Playlist Sort By Video Duration by @Hoverth in https://github.com/FreeTubeApp/FreeTube/pull/5627

## Fixes
* Don't make requests when importing subscriptions by @ChunkyProgrammer in https://github.com/FreeTubeApp/FreeTube/pull/5617
* Update playlist import to only add duplicate playlist items if the playlist already contained duplicates by @PikachuEXE in https://github.com/FreeTubeApp/FreeTube/pull/5783
* Fix the shorts tab being empty on the Local API by @absidue in https://github.com/FreeTubeApp/FreeTube/pull/5679
* Fix persistent cache incorrectly having higher priority then auto fetch feed option by @PikachuEXE in https://github.com/FreeTubeApp/FreeTube/pull/5743
* Fix video start time handling by @absidue in https://github.com/FreeTubeApp/FreeTube/pull/5719
* Fix long comments overflowing and breaking the layout by @absidue in https://github.com/FreeTubeApp/FreeTube/pull/5774
* Local API: Fix error when a community post has no likes by @absidue in https://github.com/FreeTubeApp/FreeTube/pull/5759
* Fix loading from search cache failing when no features are selected by @absidue in https://github.com/FreeTubeApp/FreeTube/pull/5775
* Fix subscriptions CSV import shortening channel names by @absidue in https://github.com/FreeTubeApp/FreeTube/pull/5793
* Fix hashtag page not using Invidious API when it should by @ChunkyProgrammer in https://github.com/FreeTubeApp/FreeTube/pull/5794
* fix `cmd+m` shortcut to minimize the window on macos and not mute the video by @jacopo-degattis in https://github.com/FreeTubeApp/FreeTube/pull/5829
* Fix saving Invidious thumbnail URLs for subscriptions by @absidue in https://github.com/FreeTubeApp/FreeTube/pull/5662
* Fix about page not reacting to locale changes by @absidue in https://github.com/FreeTubeApp/FreeTube/pull/5393
* Do not overwrite cache entries when hitting RSS ratelimits by @absidue in https://github.com/FreeTubeApp/FreeTube/pull/5756
* Fix playback of VOD/non-live videos with Invidious by @absidue in https://github.com/FreeTubeApp/FreeTube/pull/5654
* Fix audio track selection by @absidue in https://github.com/FreeTubeApp/FreeTube/pull/5697
* Properly handle Arabic hashtags by @absidue in https://github.com/FreeTubeApp/FreeTube/pull/5545
* Fix a few memory leaks while tearing down the player by @absidue in https://github.com/FreeTubeApp/FreeTube/pull/5698
* Lowercase hls manifest type to match the parser name shaka player registers by @MarmadileManteater in https://github.com/FreeTubeApp/FreeTube/pull/5648
* Fix wrong live stream playlist used for subscriptions by @ChunkyProgrammer in https://github.com/FreeTubeApp/FreeTube/pull/5876
* Fix no comment detection by @PikachuEXE in https://github.com/FreeTubeApp/FreeTube/pull/5879
* Fix subscriptions double loading on launch by @absidue in https://github.com/FreeTubeApp/FreeTube/pull/5930
* Fix seek bar thumbnail overflowing when exiting full screen by @absidue in https://github.com/FreeTubeApp/FreeTube/pull/5929
* Fix subscription tab hiding not working on launch by @absidue in https://github.com/FreeTubeApp/FreeTube/pull/5925
* Set download links on Watch even if there are no caption links by @MarmadileManteater in https://github.com/FreeTubeApp/FreeTube/pull/5916

## Changes
* Bump electron from 31.1.0 to 32.2.2
* Change the Proxy Videos Through Invidious setting to only apply when using the Invidious API by @absidue in https://github.com/FreeTubeApp/FreeTube/pull/5758
* Remove "Force Local Backend for Legacy Formats" setting by @absidue in https://github.com/FreeTubeApp/FreeTube/pull/5650
* Revert "Larger search/URL bar" by @efb4f5ff-1298-471a-8973-3d47447115dc in https://github.com/FreeTubeApp/FreeTube/pull/5773
* Settings menu & UX improvements by @kommunarr in https://github.com/FreeTubeApp/FreeTube/pull/5029
* Allow users to pass their own hardware acceleration flags on the command line on Linux by @xsmile in https://github.com/FreeTubeApp/FreeTube/pull/5438
* Update icon for channels button on side nav by @PikachuEXE in https://github.com/FreeTubeApp/FreeTube/pull/5273
* Display a better error message on the watch page when YouTube has blocked the IP address by @absidue in https://github.com/FreeTubeApp/FreeTube/pull/5757
* Change Sponsorblock Default Category Colors by @Neuheit in https://github.com/FreeTubeApp/FreeTube/pull/5686
* Update RSS tooltip by @efb4f5ff-1298-471a-8973-3d47447115dc in https://github.com/FreeTubeApp/FreeTube/pull/5753
* Clarify how to add subscriptions by @efb4f5ff-1298-471a-8973-3d47447115dc in https://github.com/FreeTubeApp/FreeTube/pull/5778
* Update subscription cache when subscribing from the channel page by @absidue in https://github.com/FreeTubeApp/FreeTube/pull/5717
* Update top nav to move query text to new window instead of remaining in old window by @PikachuEXE in https://github.com/FreeTubeApp/FreeTube/pull/5541
* Fix incorrect/translated placeholders in locale files by @absidue in https://github.com/FreeTubeApp/FreeTube/pull/5628
* Update subscription details from search results by @absidue in https://github.com/FreeTubeApp/FreeTube/pull/5795
*  Various performance improvements by @absidue in https://github.com/FreeTubeApp/FreeTube/pull/5522, https://github.com/FreeTubeApp/FreeTube/pull/5666, https://github.com/FreeTubeApp/FreeTube/pull/5891, https://github.com/FreeTubeApp/FreeTube/pull/5927 and https://github.com/FreeTubeApp/FreeTube/pull/5722
* Various code cleanups by @absidue in https://github.com/FreeTubeApp/FreeTube/pull/5520, https://github.com/FreeTubeApp/FreeTube/pull/5776, https://github.com/FreeTubeApp/FreeTube/pull/5663 and https://github.com/FreeTubeApp/FreeTube/pull/5625
* Remove mentioning of bypassing age restricted content by @efb4f5ff-1298-471a-8973-3d47447115dc in https://github.com/FreeTubeApp/FreeTube/pull/5902
* Set minimum width and height for window by @kommunarr in https://github.com/FreeTubeApp/FreeTube/pull/5931

## Known Issues
When using the Local API or the Invidious API it is possible that you will run into issues due to YouTube IP blocks. Please try to use a different Invidious Instance, VPN or proxy when you encounter those issues.

## New Contributors
* @xsmile made their first contribution in https://github.com/FreeTubeApp/FreeTube/pull/5438
* @Neuheit made their first contribution in https://github.com/FreeTubeApp/FreeTube/pull/5686
* @jacopo-degattis made their first contribution in https://github.com/FreeTubeApp/FreeTube/pull/5829
* @aliaksandr-trush made their first contribution in https://github.com/FreeTubeApp/FreeTube/pull/5841
* @Hoverth made their first contribution in https://github.com/FreeTubeApp/FreeTube/pull/5627

**Full Changelog**: https://github.com/FreeTubeApp/FreeTube/compare/v0.21.3-beta...v0.22.0-beta

**Comment by luke-bravenboer** [comment]
This might be a dumb question because of the whole overhaul, but - is there a way to use the old player UI without fully downgrading, by any chance?

**Reply by absidue** [reply]
No, that is not possible.

**Reply by ross-rosario** [reply]
Would also prefer rolling back to video.js since the new player brings solutions to problems I never had + it's objectively not intuitive.

**Reply by kommunarr** [reply]
You can create a fork or keep using older app versions if you like, but we as volunteers do not have the bandwidth to support multiple players going forward. We strive to act on each bit of actionable feedback we receive as much as possible, and unfortunately, nonspecific overarching critiques like that do not give us much to work with.

**Reply by ross-rosario** [reply]
Fair enough. I'm thankful that freetube exists however the player is. With that said, if it's not me, the someone else will make similar remarks because the old player was fine. It would help if the potential advantages of the new player are communicated a bit more comprehensively. Will it, for example, help maintaining the picture-in-picture for the next video in a playlist? Will it help maintaining theater mode? Does it make the freetube codebase a lot cleaner and more maintainable? ...etc

**Reply by kommunarr** [reply]
[You can see here](https://github.com/FreeTubeApp/FreeTube/pull/4978) just some of the bugs with the old player that were fixed by the migration. We actually have an open PR for the specific feature you mention (#5903) that was made easier by the new player.

**Reply by ross-rosario** [reply]
Well, I do appreciate this change a lot more, now that I see the number of issues being fixed and the huge code reduction it resulted in. Thanks for answering my question and for @absidue's contribution.

**Comment by Karalynx** [comment]
The big pause icon in the middle of the player is just annoying and gets in the way, I just reverted back to the previous flatpak version.

**Reply by absidue** [reply]
You can hide it with the same setting as in older versions of FreeTube: `Display Play Button In Video Player` in the `Player` section of the setting.

**Reply by MarkusGaugg** [reply]
I think it might be useful to disable the big pause button by default.

**Reply by Nickleaton** [reply]
absidue - thanks works a treat!

**Comment by CL7F** [comment]
Hi, thank you for all your work! Did it lose the captions customization?

**Reply by efb4f5ff-1298-471a-8973-3d47447115dc** [reply]
Yes it was removed, captions customization will come back eventually though!

**Reply by qupear** [reply]
> Yes it was removed, captions customization will come back eventually though!

Hope eventually not means long time. For me large captions are very important (.
Also hope to have ability to set preferred captions language. Not sure why, but previous versions were showing the language once chosen. New version seems to ignore this.

**Reply by efb4f5ff-1298-471a-8973-3d47447115dc** [reply]
> Not sure why, but previous versions were showing the language once chosen. New version seems to ignore this.

@qupear could you elaborate some more on this?

**Reply by taivlam** [reply]
@qupear Do you mean what was described in #1444?  (I created that GH issue about 3 years ago.)

**Reply by qupear** [reply]
> @qupear Do you mean what was described in #1444? (I created that GH issue about 3 years ago.)

On exactly same video Freetube 0.22 automatically loads translated captions. 0.21.3 loads just auto-generated captions with correct language. And even if I choose auto-generated captions without translation, next video again will be with translated captions. All the settings are exactly same for 0.22 and 0.21.

**Reply by absidue** [reply]
@qupear The logic didn't change it still picks the first one in the list, it never remembered the selected one between videos. The only difference is that we fixed the broken subtitle translation feature, so now it correctly shows up on all videos that don't have a subtitle track that matches your display language. So of course the translated track is going to get sorted higher in the subtitle list because it matches your display language.

**Comment by Nopenotdoingint** [comment]
Is it possible to get the red progress bar back at the bottom? It was the main reason for me using Freetube.

**Reply by Wolferey** [reply]
Echoing this, the new one looks nice but overall it's a net negative in functionality when it's inside the player and hides away, specially with the change to the playback speed button.

**Reply by efb4f5ff-1298-471a-8973-3d47447115dc** [reply]
@Nopenotdoingint what exactly do you mean with back to the bottom?

**Reply by Nopenotdoingint** [reply]
The red progress bar/playback bar on the video. It was visible (always) at the bottom of the video. It's very useful for longer form videos.

**Reply by efb4f5ff-1298-471a-8973-3d47447115dc** [reply]
Im sorry to inform you that this wont come back.

**Reply by Hoverth** [reply]
Is there any design reason why, or is it just a technical limitation of using Shaka? I don't see any large difficulty in just putting a progress-bar type element underneath the video player, and then allowing the enabling/disabling of it. 

The progress bar being there underneath the video was a great tool for accessibility, with not being able to check how far through a video you are without interacting with the player being, from what I can see, the main issue here.

**Reply by qupear** [reply]
Definitely some progress indicator for video is needed. It could be even just a textbox with passed time.

**Reply by efb4f5ff-1298-471a-8973-3d47447115dc** [reply]
This is shaka and not us. We are collecting as much community feedback as we can to iterate on the player. 

We will re-evaluate if there is something that can be done here either by us or upstream.

**Reply by taivlam** [reply]
I also found the prior progress bar useful, as @Nopenotdoingint described it.

**Comment by chriscjcj** [comment]
Thank you for all of your incredible work on this app. I am grateful!

This comment is regarding the Mac version. I have not tested this behavior on other platforms.

The keystroke for FreeTube's back button has been alt-left arrow. I think that shaka-player uses this keystroke to go back a few frames in the video. If the video itself has focus, al-left arrow doesn't invoke the back button. You have to click on another part of FreeTube's UI first and then alt-left arrow will function as before.

If I may recommend, it's always bugged me a bit that alt-left arrow was the keystroke for the back button anyway. On the mac, Command-left bracket is the most common way to invoke the back button. This keystroke currently does noting in FreeTube, near as I can tell. 

Perhaps moving the back and forward function to command-left bracket and command-right bracket would solve both problems.

**Reply by absidue** [reply]
We definitely would not want to remove a keyboard shortcut that people have got used to, especially considering that it is the correct one for Windows and Linux, however it should be possible to add extra ones that we only activate on macOS.

Please open a feature request for those two extra macOS-only keyboard shortcuts if you want to see them added to FreeTube, please also mention in the feature request which characters you mean `(`, `[` or `{`.

**Reply by chriscjcj** [reply]
[Done and done](https://github.com/FreeTubeApp/FreeTube/issues/5966). Thanks so much.

**Comment by SyniHD** [comment]
Thank you so much for the update. Videos feel like they start faster after clicking them as well. I would just like to request that the resolution button on the video match the theme. I'm using Dark Mode and clicking the resolution button shows me the available options on a white background.

**Reply by efb4f5ff-1298-471a-8973-3d47447115dc** [reply]
Good piece of feedback. We will definitely work on this!

**Comment by ruggeriruggero** [comment]
Playback speed icon is a fixed one now instead of showing the current playback speed... this is a nitpick but I liked seeing what's the current playback speed as I like to change it often while I'm watching, through the keyboard hotkeys, so I need some visual feedback.

**Reply by absidue** [reply]
We have an open feature request here for showing a HUD/temporary overlay when the volume is changed by means other than the volume slider, maybe that feature request could be extended to treat the playback rate in a similar way.

https://github.com/FreeTubeApp/FreeTube/issues/4658

**Reply by ruggeriruggero** [reply]
yeah that would be even better

**Reply by qupear** [reply]
I also like to be able to change playback speed by clicking speed icon as in older versions. Hotkeys are good but not all devices have those keys. Previously I was able to freetube with only mouse or remote control.

**Reply by absidue** [reply]
> I also like to be able to change playback speed by clicking speed icon as in older versions.

@qupear That was intentionally removed because it was a major accessibility issue, as it made it impossible to open the playback rate menu unless you were using a mouse to hover over the button, touch screen users (e.g. mobile phones) couldn't use it at all.

You can still use FreeTube with just a mouse, you just need to click to open the menu like with all other player buttons that have menus.

**Reply by qupear** [reply]
> touch screen users (e.g. mobile phones) couldn't use it at all.

Thanks now I get it, sounds reasonable.

**Reply by pycoed** [reply]
I find the same problem with the resolution button which does not now show what is current: for those of us with poor broadband & variable 4G signals, it's very useful to be able to change resultion on the fly to prevent buffering. I could do this with 0.21 but no longer?

**Comment by GabLnrs** [comment]
Do you know why the age restricted videos arent playable anymore?

**Reply by absidue** [reply]
YouTube patched the age-restriction bypass (they weren't forcing logging in in one specific place, now they are forcing logging in for age-restricted videos everywhere), now the only watch to watch age-restricted videos is in the official apps or website with an age-verified YouTube account.

**Reply by GabLnrs** [reply]
Pretty sad, knowing that i cant login with my youtube account directly on freetube. But thank you because i couldnt find any response.

**Reply by absidue** [reply]
Logging into FreeTube with a Google account would defeat the whole reason for FreeTube's existence, to be a more privacy friendly way of using YouTube. It would also put your YouTube account at risk, because you would be using it outside of the official YouTube apps.

**Reply by k0zyrev** [reply]
It still would be a great optional feature to use only on restricted videos. In terms of privacy there is no difference if you'd access it via Freetube or any other way. It also should be only used to access restricted videos. And users don't have to accept the risk if they don't want to, but for those who do it would be a great option, still better than using youtube in a browser.

**Comment by ccl671** [comment]
Videos in the shorts tab still aren't showing for me in the latest release  - [v0.22.0 Beta].
Is it just me and if so, why ?
Is it something to do with Local API ?
Thanks

**Reply by absidue** [reply]
We were only able to fix it for the local API, as for the Invidious API it needs to be fixed on the Invidious side. However once it is fixed on the Invidious side, it should start working automatically in FreeTube without any changes needed on the FreeTube side.

For the moment you can switch to the local API if you want to see content on the channel shorts tab.

**Reply by ccl671** [reply]
Thanks for the info

**Comment by MrBelles** [comment]
Seeing the other posts on this thread, I second the request for an alternative element to the always-on progress bar, and visible playback speed (and resolution) on hover.

In 0.21.3 I was able to play a video PIP, then navigate around the app in the same window, but now the PIP closes when I navigate, and this feature was useful when preparing to select the next video towards the end.

**Reply by absidue** [reply]
@MrBelles PiP was never meant to stay open longer than while you were on the watch page, the fact that it did was a bug and also caused various bugs such as not saving your watch progress and sponsor block skipping not working.

**Reply by MrBelles** [reply]
Understandable, thanks for the explanation, I agree that behavior came with bugs and is best to keep the new behavior.

**Comment by basedBSD** [comment]
THANKS A LOT FOR STABLE VOLUME OPTION!
idk if this is the version that introduced it - I don't update my software often - whoever added it, it was a dream option for at least a few years now

**Reply by efb4f5ff-1298-471a-8973-3d47447115dc** [reply]
You can thank @absidue for that :)

**Reply by basedBSD** [reply]
@absidue ur a legend, really helps me during the night

**Comment by yeskunall** [comment]
Hey, thanks for all the hard work on this @absidue. I know the amount of work that can go into maintaining a video player, and I’m not questioning the move to Shaka Player—it looks great!—as long as we keep the ability to open videos in an external player.

__PS__: On a side note, `⌘+,` no longer opens FreeTube’s settings page when the player is loaded (or in view, it seems). Instead, `⌘+,` now loads (buffer, not play) the video instead.

FreeTube Version 0.22.0 (0.22.0)

**Reply by PikachuEXE** [reply]
Fix PR opened: https://github.com/FreeTubeApp/FreeTube/pull/6054

**Comment by exefile2** [comment]
Is there any way to downgrade to the previous player ui? After the update, the app is putting so much load in my pc (low end) and its now difficult to use freetube. Even playing 480p is difficult for my pc. Kindly help...

**Reply by absidue** [reply]
@exefile2 You can downgrade to the previous version of FreeTube for the moment, presumably your computer is to old to handle the newer video codecs that FreeTube currently uses.

**Reply by exefile2** [reply]
From where can I downgrade that? And yes, my pc is a low end pc.

**Reply by absidue** [reply]
https://github.com/FreeTubeApp/FreeTube/releases

**Comment by Emmitt222** [comment]
I'm getting a message saying "There are no comments available for this video" instead of the actual comments on most videos. Didn't have any problems with this before upgrading to this version.

**Reply by absidue** [reply]
@Emmitt222 Are you sure you are using 0.22.0 and not still on 0.21.3? As that was fixed in 0.22.0.

**Comment by hueldoeu** [comment]
freetube is for windows what newpipe is for android. i really like what you have done.

**Comment by k0zyrev** [comment]
I'm trying the current [nightly build ](https://github.com/FreeTubeApp/FreeTube/actions/runs/12268233241)because of the recent "Unexcepted token [1:71]" errors, and I've noticed that if I change playback speed and then pause the video, the speed reverts to the default value. Clearly a bug.

**Reply by k0zyrev** [reply]
in addition to that, my default pb rate is x2 and I can't even manually select x1 - when I click 1, it just hops back to 2

**Reply by k0zyrev** [reply]
but if I change the playback speed with keyboard shortcuts, everything works as intended - I can pause without speed being changed, I can play at x1 even if the default is x2

**Reply by absidue** [reply]
Next time please search in the issue tracker before spamming loads of people with 3 messages (everyone that has commented on this thread gets automatically signed up to notifications for all future comments): https://github.com/FreeTubeApp/FreeTube/issues/6281

**Comment by hcgrove** [comment]
Earlier today I upgraded freetube to version 0.22.0 using the Debian package (I run Debian 12.8) downloaded from [freetube.io](http://freetube.io/), but ever since freetube has not wanted to play a single movie.

When I click on a movie in a my subscriptions, everything looks normal (i.e. a list of movies on the right, information below the movie), but on top of the movie splash, there is a big white (partly transparent) circle, with a black play icon, with a little back indication going around the perimeter of the circle - I guess that's how it's supposed to look with the new player.
 
But nothing happens and nothing (in freetube) responds to either clicks nor keypresses. 

On most attempts I just did a `killall freetube`, but on one of the later attempts I let it run and discovered that after a while (man minutes), freetube's window turned black and it started responding to keypresses (well, ctrl+q at least).

I've tried with several movies, and somewhere among the tries I rebooted to try to clean out everything.

Someone said the new player causes higher load and until the window turns black one process uses +100% cpu and several GB of memory.

For now I've downgraded to 0.21.3 (just installing the old debian package) and that seems to work. But (while it might take some days before I get it done) I can try upgrading again and collect some debugging information if that is needed. As long as it's possible to downgrade if the new version doesn't start to work.
